src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +122,43 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            if (count <= 0)
+            {
+                return source.Skip(0);
```

Comment:
Is the choice of `Skip(0)` here arbitrary (as opposed to any other operator which would return the original sequence)? Do we refrain from returning `source` because we want some form of hiding of the source here?